python/cac_tripplanner/templates/partials/event-detail.html

```diff
@@ -21,28 +21,77 @@
         <div class="info-article-header-info">
             <div class="info-article-header-main">
                 <h2 class="info-article-title">{{ event.name }}</h2>
-                {% if event.destination %}
+                {% if event.first_destination %}
                 <h3 class="info-event-destination">
                     at
-                    <a class="info-event-destination-link"
-                        href="{% url 'place-detail' pk=event.destination.pk %}">
-                        {{ event.destination.name }}
+                    <!-- limit to the first four destinations, with JS to show the rest, if any -->
+                    {% with max_destinations=4 %}
+                    {% for event_dest in event.event_destinations.all %}
+                    <div class="{% if forloop.counter > max_destinations %}extra-destination hidden-extra{% endif %}">
+                        <a class="info-event-destination-link"
+                            href="{% url 'place-detail' pk=event_dest.destination.pk %}">
+                            {{ event_dest.destination.name }}
+                        </a>
+                        {% if event_dest.start_date and event_dest.end_date %}
+                        {% if event_dest.single_day %}
+                        <!-- only show dates for multi-day events -->
```

Comment:
There are a number of possible combinations of dates and times:
- Single-day event (for which the destinations obviously should be single-day as well)
- Multi-day destination on a multi-day event
- Single-day destination on a multi-day event

Here's an example of "everything on one day":
![image](https://user-images.githubusercontent.com/6598836/65275649-16266080-daf4-11e9-9091-bb6d96d7fe97.png)

Nothing there is wrong or confusing, but the dates on the destinations are superflous--it could just show times.

Here's a multi-day event with one single-day and one multi-day destination:
![image](https://user-images.githubusercontent.com/6598836/65275760-4bcb4980-daf4-11e9-95f7-7963f3df2883.png)

This, too, is not incorrect but might look better if the single-day destination used the format that's being used for single-day events, i.e.
> Thu Sep 12
> 6pm - 8pm

If the current setup is what's desired, that's fine--each of the labels is correct and complete.  Some of them are just potentially a little less clear than they could be without repeated bits.

(Also, this "only show dates for multi-day events" comment doesn't quite match what's happening around it.)

Reply:
The `single_day` Django helper method was returning inconsistent results because it wasn't timezone aware (it depended on the time range whether it seemed to work or not). It was misbehaving similarly for the overall event date/time range as well.